packages/http-client/src/http-request-config.ts

```diff
@@ -40,3 +46,90 @@ const defaultDisallowedKeys = [
   'headers',
   'params'
 ];
+
+/**
+ * Build [[HttpRequestConfig]] from [[HttpRequestConfigWithOrigin]]
+ * @param configWithOrigin - An http request config with origin information
+ * @returns The resulting [[HttpRequestConfig]]
+ * @internal
+ */
+export function buildHttpRequestConfig(
+  configWithOrigin: HttpRequestConfigWithOrigin
+): HttpRequestConfig {
+  const requestConfig = configWithOrigin as HttpRequestConfig;
+  requestConfig.headers = getOptionWithPriority(configWithOrigin.headers);
+  requestConfig.params = getOptionWithPriority(configWithOrigin.params);
+  return requestConfig;
+}
+
+/**
+ * Build http request options from given options with origin information. When reaching conflicts, values with higher priority are chosen.
+ * @param headersOrParams - The given options with origin information
+ * @returns The resulting options
+ * @internal
+ */
+export function getOptionWithPriority(
+  headersOrParams?: Record<string, ValueWithOrigin>
+): Record<string, string> | undefined {
+  if (headersOrParams) {
+    return Object.entries(headersOrParams).reduce(
+      (prev, [key, valueWithOrigin]) => {
+        const value = getValueWithPriority(valueWithOrigin);
+        if (value) {
+          prev[key] = value;
+        }
+        return prev;
+      },
+      {} as Record<string, string>
+    );
+  }
+}
+
+/**
+ * Pick the value with highest priority, defined by origin.
+ * @param valueWithOrigin - Values with origin information
+ * @returns The value with highest priority.
+ * @internal
+ */
+export function getValueWithPriority(
+  valueWithOrigin: ValueWithOrigin
+): string | undefined {
+  return (
+    valueWithOrigin.Custom ||
+    valueWithOrigin.DestinationProperty ||
+    valueWithOrigin.Destination ||
+    valueWithOrigin.RequestConfig
+  );
+}
+
+/**
+ * Merge all the options to one object with the origin information
+ * @param optionsWithOrigin - One or multiple objects that contains http request options with origin information
+ * @returns The merged object
+ * @internal
+ */
+export function mergeOptionsWithOrigin(
+  ...optionsWithOrigin: OptionWithOrigin[]
+): Record<string, ValueWithOrigin> {
+  return optionsWithOrigin.reduce(
+    (prevOptionWithOrigin, currentOptionWithOrigin) => {
+      if (currentOptionWithOrigin.option) {
+        Object.entries(currentOptionWithOrigin.option).reduce(
+          (prev, [optionKey, optionValue]) => {
+            if (prev[optionKey]) {
+              prev[optionKey][currentOptionWithOrigin.origin] = optionValue;
```

Comment:
In this method also some merging takes place so the order of the arguments needs to be in line with the priority.